### DISCUSSION TITLE
GitHub Actions: Integrate Coverity Scan?

### DISCUSSION THREAD

**Discussion by jumbi77** [openzfs/zfs#16580] [General]
Hi guys,

OpenZFS is already registered on [coverity](https://scan.coverity.com/projects/openzfs-zfs), but (as far as I can tell) currently an Admin must submit a new build manually. I think coverity can be automated with GitHub actions as a new workflow [1]. Would this be a useful or is (a) a manual/sporadically analysis enough and/or (b) GitHub actions, with all current workflows, already pretty "busy" and is taking already too long?

[1] [https://sig-product-docs.synopsys.com/bundle/bridge/page/documentation/c_using-synopsys-github-action.html](https://sig-product-docs.synopsys.com/bundle/bridge/page/documentation/c_using-synopsys-github-action.html)

**Comment by robn** [comment]
I can't speak to the CI side, but I do have an admin bit on the Coverity project and mostly I find it not especially useful, at least not at this point. Mostly, its very difficult to express a lot of the safety rules we have, and its UI is awful, so there's a ton of stuff in there that isn't relevant and/or is very hard triage.

I've been experimenting with [CodeChecker](https://github.com/Ericsson/codechecker) a bit lately, which for our purposes is mostly a nice wrapper around Clang Analyzer, clang-tidy and cppcheck. It's also not perfect, but is a bit easier to work with and the UI seems a lot more useful. I will likely end up running this regularly myself, probably at least an automated daily run from master, but regardless of whether or not the project ended up running something, I would be keeping my own because I have private customer work that I'd like to receive the same treatment.
